### ISSUE TITLE
Create Ore Generation Incompatibility

### ISSUE THREAD

**Issue by The-Math-Fish** [EwyBoy/OreTweaker#9]
I apologize for the length of this, but I couldn't upload config files directly into here.

It appears that at OreTweaker and Create ore generation are incompatible with one another. I kept Create's ore generation as is and only edited the vanilla generation using OreTweaker. Here are copies of the Create and OreTweaker configs:

**Create**
```
[worldgen]

	#
	#Modify Create's impact on your terrain
	[worldgen.v2]
		#
		#Prevents all worldgen added by Create from taking effect
		disableWorldGen = false
		#
		#Forward caught TileEntityExceptions to the log at debug level.
		logTeErrors = false

		[worldgen.v2.copper_ore]
			#
			#Range: > 0
			clusterSize = 18
			#
			#Amount of clusters generated per Chunk.
			#  >1 to spawn multiple.
			#  <1 to make it a chance.
			#  0 to disable.
			#Range: 0.0 ~ 512.0
			frequency = 2.0
			#
			#Range: > 0
			minHeight = 40
			#
			#Range: > 0
			maxHeight = 85

		[worldgen.v2.weathered_limestone]
			#
			#Range: > 0
			clusterSize = 128
			#
			#Amount of clusters generated per Chunk.
			#  >1 to spawn multiple.
			#  <1 to make it a chance.
			#  0 to disable.
			#Range: 0.0 ~ 512.0
			frequency = 0.015625
			#
			#Range: > 0
			minHeight = 10
			#
			#Range: > 0
			maxHeight = 30

		[worldgen.v2.zinc_ore]
			#
			#Range: > 0
			clusterSize = 14
			#
			#Amount of clusters generated per Chunk.
			#  >1 to spawn multiple.
			#  <1 to make it a chance.
			#  0 to disable.
			#Range: 0.0 ~ 512.0
			frequency = 4.0
			#
			#Range: > 0
			minHeight = 15
			#
			#Range: > 0
			maxHeight = 70

		[worldgen.v2.limestone]
			#
			#Range: > 0
			clusterSize = 128
			#
			#Amount of clusters generated per Chunk.
			#  >1 to spawn multiple.
			#  <1 to make it a chance.
			#  0 to disable.
			#Range: 0.0 ~ 512.0
			frequency = 0.015625
			#
			#Range: > 0
			minHeight = 30
			#
			#Range: > 0
			maxHeight = 70

		[worldgen.v2.dolomite]
			#
			#Range: > 0
			clusterSize = 128
			#
			#Amount of clusters generated per Chunk.
			#  >1 to spawn multiple.
			#  <1 to make it a chance.
			#  0 to disable.
			#Range: 0.0 ~ 512.0
			frequency = 0.015625
			#
			#Range: > 0
			minHeight = 20
			#
			#Range: > 0
			maxHeight = 70

		[worldgen.v2.gabbro]
			#
			#Range: > 0
			clusterSize = 128
			#
			#Amount of clusters generated per Chunk.
			#  >1 to spawn multiple.
			#  <1 to make it a chance.
			#  0 to disable.
			#Range: 0.0 ~ 512.0
			frequency = 0.015625
			#
			#Range: > 0
			minHeight = 20
			#
			#Range: > 0
			maxHeight = 70

		[worldgen.v2.scoria]
			#
			#Range: > 0
			clusterSize = 128
			#
			#Amount of clusters generated per Chunk.
			#  >1 to spawn multiple.
			#  <1 to make it a chance.
			#  0 to disable.
			#Range: 0.0 ~ 512.0
			frequency = 0.03125
			#
			#Range: > 0
			minHeight = 0
			#
			#Range: > 0
			maxHeight = 10
```

**OreTweaker**
```
"Enable Debug Output" = false
"Disable Ores" = ["minecraft:coal_ore", "minecraft:iron_ore", "minecraft:gold_ore", "minecraft:diamond_ore", "minecraft:lapis_ore", "minecraft:redstone_ore", "minecraft:emerald_ore"]

[["Custom Ore"]]
	"Ore Name" = "minecraft:lapis_ore"
	"Max Vein Size" = 7
	"Filler Name" = "minecraft:stone"
	"Min Vein Level" = 1
	"Max Vein Level" = 48
	"Spawn Rate" = 8

[["Custom Ore"]]
	"Ore Name" = "minecraft:redstone_ore"
	"Max Vein Size" = 7
	"Filler Name" = "minecraft:stone"
	"Min Vein Level" = 1
	"Max Vein Level" = 48
	"Spawn Rate" = 8

[["Custom Ore"]]
	"Ore Name" = "minecraft:emerald_ore"
	"Max Vein Size" = 1
	"Filler Name" = "minecraft:stone"
	"Min Vein Level" = 1
	"Max Vein Level" = 48
	"Spawn Rate" = 1
```

With these configs I used World Stripper to uncover the terrain in new chunks and this is what I got:

![image](https://user-images.githubusercontent.com/82321657/115957628-9a5f7780-a4d1-11eb-8b23-c31fbc809d65.png)

You may notice that EvilCraft is also in this mod. OreTweaker does not interfere with its ore generation. I did a similar thing in The Nether as I have Netherrocks installed as well (the only other mod that affects ore generation) and it appears to work as intended:

![image](https://user-images.githubusercontent.com/82321657/115957880-faa2e900-a4d2-11eb-80d3-023cb4eb3880.png)

So this seems to be specific to Create. As I have no way of telling whether this is a Create or an OreTweaker problem, I am simply reporting this to both. And for the sake of completion, here is my log file for this session:

[latest.log](https://github.com/EwyBoy/OreTweaker/files/6369529/latest.log)

**Comment by The-Math-Fish** [comment]
Note that I do have a workaround I will implement to fix this: Namely disable Create's ore generation entirely and add the "ores" as custom ores in OreTweaker. So, while this isn't a major problem, it is definitely something to look into.

**Comment by EwyBoy** [comment]
My version of Ore Tweaker does not currently support anything beyond Minecraft 1.12, but some total retard thought it was a good idea to create a mod with the exact same behavior and name so I am getting flodded with issue reports I have nothing to do with. I am releasing my version of Ore Tweaker for 1.16 later this week.

You prob want to contact this guy here: https://github.com/noahc3/oretweaker/issues

I am gonna make sure this works out of the box once I release my 1.16 version ;-)

**Comment by The-Math-Fish** [comment]
Alright. In this case I will be removing this guy's mod and replacing it with yours as soon as it is out. In addition, I will be adding to the description of my modpack that there was once an illegitimate version of OreTweaker present and link the site.

Thank you for your time. You can close this issue as this isn't an issue with your mod, specifically.

**Comment by EwyBoy** [comment]
I want to state that this person has done nothing wrong and his version is not illegitimate. All my mods are open source under the MIT License and everyone can create their own version of my mods. This person has written his mod from scratch but the main problem I have is that he is distributing it under the exact same name as my mod but without a space between the two words.
This causes a lot of confusion for the end users and a ton of people let to believe they are using my mod that haven't been updated since 1.12. The result is me getting a ton of bugs and issues that I cannot fix and needs to be redirected. It is common in the modded Minecraft community to change the name of the mod once you create a continuation / successor of someone product.
Like "Too Many Items" (TMI) became "Not Enough Item" (NEI) and then "Just Enough Item" (JEI).

**Comment by EwyBoy** [comment]
Hey, I just released Ore Tweaker 2.0.0 on CurseForge.
I have "fixed" the bug you where experiencing but I am still trying to figure it properly out.

There seem to be a issue where the game thinks creates ores all are "minecraft:lapis" for some strange reason.
So when you tell the mod to disable lapis it also disables all Create ores..
I have created a temp workaround to fix this by checking that if create is present it will not tweak lapis. This also results in that you cannot tweak lapis but is a temporary compromise until I can properly figure out what going on. I might have to check in with the creators of Create and check if they inherit some properties from Lapis in their mod.

Link to new 1.16 version of Ore Tweaker :: https://www.curseforge.com/minecraft/mc-mods/ore-tweaker
